src/librustc_mir/transform/const_prop.rs

```diff
@@ -580,8 +580,20 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
 
                 return None;
             }
+            Rvalue::ThreadLocalRef(def_id) => {
+                trace!("skipping ThreadLocalRef({:?})", def_id);
 
-            _ => {}
+                return None;
```

Comment:
This looks like a change in behavior?

Reply:
So I think this used to be covered by the `Ref(..)` case before we split thread locals into their own `Rvalue`. I could be wrong about that.

Reply:
We used to catch this later, so it's not a change in behaviour, just a hypothetical perf improvement for code that uses thousands of thread locals (unlikely).

Still it's a good thing to document this here properly